<!-- HTML header for doxygen 1.8.8-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=9"/>
    <meta name="generator" content="Doxygen 1.8.13"/>
    <title>trimAl: Alignment Class Reference</title>
    <link href="tabs.css" rel="stylesheet" type="text/css"/>
    <script type="text/javascript" src="jquery.js"></script>
    <script type="text/javascript" src="dynsections.js"></script>
    <link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
    <link href="doxygen.css" rel="stylesheet" type="text/css" />
    <link href="CustomExtra.css" rel="stylesheet" type="text/css"/>
    <script type="text/javascript">
        function myFunction(ID) {
            const divs = document.getElementsByName(ID);
            for (let x of divs) {
                if (x.style.display === "none") {
                    x.style.display = "block";
                } else {
                    x.style.display = "none";
                }
            }
        } 
    </script>
</head>
<body>
    <div id="top"><!-- do not remove this div, it is closed by doxygen! -->
        <div id="titlearea">
            <table cellspacing="0" cellpadding="0" width=100%>
                <tbody>
                    <tr style="height: 12px;">
                        <td id="projectlogo"><img alt="Logo" src="Logo.png"/></td>
                    </tr>
                </tbody>
            </table>
            <div class="header">
                <br>
            </div>
        </div>
        <!-- end header part -->
    <!-- Generated by Doxygen 1.8.13 -->
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('class_alignment.html','');});
</script>
<div id="doc-content">
<div class="header">
  <div class="summary">
<a href="#nested-classes">Classes</a> &#124;
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="#pri-attribs">Private Attributes</a> &#124;
<a href="class_alignment-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">Alignment Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Class containing an alignment<br />
 This class stores the alignment sequences with it's names, residues and extra information.<br />
 It contains multiple methods regarding the sequences.<br />
 It also contains submodules that provide methods for<b> Calculating statistics</b>, <b> Trim the alignment</b> and<b> Printing sequences information</b>.  
 <a href="class_alignment.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="_alignment_8h_source.html">Alignment.h</a>&gt;</code></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Classes</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">class &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment_1_1sequences_matrix.html">sequencesMatrix</a></td></tr>
<tr class="memdesc:"><td class="mdescLeft">&#160;</td><td class="mdescRight">Internal <a class="el" href="class_alignment.html" title="Class containing an alignment  This class stores the alignment sequences with it&#39;s names...">Alignment</a> Class that represents a sequences matrix.  <a href="class_alignment_1_1sequences_matrix.html#details">More...</a><br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a73e7500f4334361430be77f8d944be7f"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a73e7500f4334361430be77f8d944be7f">fillMatrices</a> (bool aligned, bool checkInvalidChars=true)</td></tr>
<tr class="memdesc:a73e7500f4334361430be77f8d944be7f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to initialize matrices and check if the sequences have been correctly loaded and are free of errors<br />
It checks if sequences contain unknown characters, sets the isAligned flag and initializes 'saveResidues' and 'saveSequences', depending on the sizes of the sequences (whether or not they have the same length).  <a href="#a73e7500f4334361430be77f8d944be7f">More...</a><br /></td></tr>
<tr class="separator:a73e7500f4334361430be77f8d944be7f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a091e21f0cfb741cb5cdc3535c6cc287f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a091e21f0cfb741cb5cdc3535c6cc287f">Alignment</a> ()</td></tr>
<tr class="memdesc:a091e21f0cfb741cb5cdc3535c6cc287f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Constructor.  <a href="#a091e21f0cfb741cb5cdc3535c6cc287f">More...</a><br /></td></tr>
<tr class="separator:a091e21f0cfb741cb5cdc3535c6cc287f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aeeee94970a445f32aa79646c249befef"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#aeeee94970a445f32aa79646c249befef">Alignment</a> (<a class="el" href="class_alignment.html">Alignment</a> &amp;)</td></tr>
<tr class="memdesc:aeeee94970a445f32aa79646c249befef"><td class="mdescLeft">&#160;</td><td class="mdescRight">Copy Constructor. <br />
 The copy constructor does not make a copy of all structures present on the original alignment.<br />
 Instead, all alignments that share a common origin, share some information, and also, a counter (<a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a>) of how many alignments share the same information.<br />
 This allows us to keep the information on memory while any alignment is using it.<br />
 When an alignment is destroyed, the counter is substracted by 1, and, if it is 0 after the substraction, the shared data is destroyed.  <a href="#aeeee94970a445f32aa79646c249befef">More...</a><br /></td></tr>
<tr class="separator:aeeee94970a445f32aa79646c249befef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8787573a5060136c2a51f40cf229e356"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a8787573a5060136c2a51f40cf229e356">~Alignment</a> ()</td></tr>
<tr class="memdesc:a8787573a5060136c2a51f40cf229e356"><td class="mdescLeft">&#160;</td><td class="mdescRight">Destructor. The destructor takes care of removing the non-shared information of each alignment when is destroyed.<br />
 It also substracts 1 from the shared counter <a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a>.<br />
 When <a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a> arrives to 0, the shared data is destroyed, including <a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a>.  <a href="#a8787573a5060136c2a51f40cf229e356">More...</a><br /></td></tr>
<tr class="separator:a8787573a5060136c2a51f40cf229e356"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af9c7aec9082ef16573d7fe37a93a638a"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#af9c7aec9082ef16573d7fe37a93a638a">getNumSpecies</a> ()</td></tr>
<tr class="memdesc:af9c7aec9082ef16573d7fe37a93a638a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Number of sequences getter.  <a href="#af9c7aec9082ef16573d7fe37a93a638a">More...</a><br /></td></tr>
<tr class="separator:af9c7aec9082ef16573d7fe37a93a638a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af5deab9f747b96c43a1004c2e1170095"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#af5deab9f747b96c43a1004c2e1170095">getSequences</a> (std::string *names)</td></tr>
<tr class="memdesc:af5deab9f747b96c43a1004c2e1170095"><td class="mdescLeft">&#160;</td><td class="mdescRight">Getter for the sequences names.  <a href="#af5deab9f747b96c43a1004c2e1170095">More...</a><br /></td></tr>
<tr class="separator:af5deab9f747b96c43a1004c2e1170095"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae8b5f6fbc7ffb828c7eb51ae74e6888c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ae8b5f6fbc7ffb828c7eb51ae74e6888c">getSequences</a> (std::string *names, int *lenghts)</td></tr>
<tr class="memdesc:ae8b5f6fbc7ffb828c7eb51ae74e6888c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Getter for the sequences names and its lenghts.  <a href="#ae8b5f6fbc7ffb828c7eb51ae74e6888c">More...</a><br /></td></tr>
<tr class="separator:ae8b5f6fbc7ffb828c7eb51ae74e6888c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4dcdeecf3fad5bb749f3f6051655e17a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a4dcdeecf3fad5bb749f3f6051655e17a">getSequences</a> (std::string *names, std::string *<a class="el" href="class_alignment.html#ab0cafdefe306c19917a8c58abd2f8918">sequences</a>, int *lenghts)</td></tr>
<tr class="memdesc:a4dcdeecf3fad5bb749f3f6051655e17a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Getter for the sequences, its names and lenghts.  <a href="#a4dcdeecf3fad5bb749f3f6051655e17a">More...</a><br /></td></tr>
<tr class="separator:a4dcdeecf3fad5bb749f3f6051655e17a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab24c9f56fb953e3e01eec2c850c6aee5"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ab24c9f56fb953e3e01eec2c850c6aee5">getSequenceNameOrder</a> (std::string *names, int *orderVector)</td></tr>
<tr class="memdesc:ab24c9f56fb953e3e01eec2c850c6aee5"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to map two sets of sequences, own and external.<br />
The method accepts a vector of names, and test them against the alignment own sequences' names.<br />
If the sequence is present, its index will be stored in orderVector.<br />
.  <a href="#ab24c9f56fb953e3e01eec2c850c6aee5">More...</a><br /></td></tr>
<tr class="separator:ab24c9f56fb953e3e01eec2c850c6aee5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5b534495c7c6fac732b290359f3b8992"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a5b534495c7c6fac732b290359f3b8992">getNumAminos</a> ()</td></tr>
<tr class="memdesc:a5b534495c7c6fac732b290359f3b8992"><td class="mdescLeft">&#160;</td><td class="mdescRight">Residues number getter. It counts gaps as residue.  <a href="#a5b534495c7c6fac732b290359f3b8992">More...</a><br /></td></tr>
<tr class="separator:a5b534495c7c6fac732b290359f3b8992"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f45aaea7bd507135890b1c8d37db930"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a9f45aaea7bd507135890b1c8d37db930">setWindowsSize</a> (int ghWindow, int shWindow)</td></tr>
<tr class="memdesc:a9f45aaea7bd507135890b1c8d37db930"><td class="mdescLeft">&#160;</td><td class="mdescRight">Windows setter.  <a href="#a9f45aaea7bd507135890b1c8d37db930">More...</a><br /></td></tr>
<tr class="separator:a9f45aaea7bd507135890b1c8d37db930"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acf3b1ef007f2903570bb2c80105f1876"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#acf3b1ef007f2903570bb2c80105f1876">setBlockSize</a> (int blockSize)</td></tr>
<tr class="memdesc:acf3b1ef007f2903570bb2c80105f1876"><td class="mdescLeft">&#160;</td><td class="mdescRight">BlockSize Setter.  <a href="#acf3b1ef007f2903570bb2c80105f1876">More...</a><br /></td></tr>
<tr class="separator:acf3b1ef007f2903570bb2c80105f1876"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab4d73aec4e1ec82bee55321791a1c6dc"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ab4d73aec4e1ec82bee55321791a1c6dc">calculateSeqOverlap</a> ()</td></tr>
<tr class="memdesc:ab4d73aec4e1ec82bee55321791a1c6dc"><td class="mdescLeft">&#160;</td><td class="mdescRight">Calculates overlap between sequences.  <a href="#ab4d73aec4e1ec82bee55321791a1c6dc">More...</a><br /></td></tr>
<tr class="separator:ab4d73aec4e1ec82bee55321791a1c6dc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:accfd82f91b9cdbe7a14ba85a1bada8cf"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#accfd82f91b9cdbe7a14ba85a1bada8cf">printSeqIdentity</a> ()</td></tr>
<tr class="memdesc:accfd82f91b9cdbe7a14ba85a1bada8cf"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to print different identity values computed from the alignment.<br />
In this method we assess the identity values matrix, as well as different average values. <br />
Moreover, the method computes which one is the most similar sequence.  <a href="#accfd82f91b9cdbe7a14ba85a1bada8cf">More...</a><br /></td></tr>
<tr class="separator:accfd82f91b9cdbe7a14ba85a1bada8cf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a57b5686aff7324890422aa4295bc62d1"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a57b5686aff7324890422aa4295bc62d1">printSeqOverlap</a> ()</td></tr>
<tr class="memdesc:a57b5686aff7324890422aa4295bc62d1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Prints the overlap between sequences.  <a href="#a57b5686aff7324890422aa4295bc62d1">More...</a><br /></td></tr>
<tr class="separator:a57b5686aff7324890422aa4295bc62d1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3f8f2eafd37c9b5bc275b9889a2de96a"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a3f8f2eafd37c9b5bc275b9889a2de96a">getAlignmentType</a> () const</td></tr>
<tr class="memdesc:a3f8f2eafd37c9b5bc275b9889a2de96a"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="class_alignment.html" title="Class containing an alignment  This class stores the alignment sequences with it&#39;s names...">Alignment</a> type getter. <br />
 See <a class="el" href="defines_8h.html#a69d5409d6e7370d6bd3b4f931702e936">SequenceTypes</a>.  <a href="#a3f8f2eafd37c9b5bc275b9889a2de96a">More...</a><br /></td></tr>
<tr class="separator:a3f8f2eafd37c9b5bc275b9889a2de96a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4fdfd8529dd7d554260377352a68a571"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a4fdfd8529dd7d554260377352a68a571">isFileAligned</a> ()</td></tr>
<tr class="memdesc:a4fdfd8529dd7d554260377352a68a571"><td class="mdescLeft">&#160;</td><td class="mdescRight">isAligned getter.  <a href="#a4fdfd8529dd7d554260377352a68a571">More...</a><br /></td></tr>
<tr class="separator:a4fdfd8529dd7d554260377352a68a571"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a57cb9cad92f31ec56982a4191d2cbce4"><td class="memItemLeft" align="right" valign="top"><a class="el" href="class_alignment.html">Alignment</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a57cb9cad92f31ec56982a4191d2cbce4">getTranslationCDS</a> (<a class="el" href="class_alignment.html">Alignment</a> *proteinAlignment)</td></tr>
<tr class="memdesc:a57cb9cad92f31ec56982a4191d2cbce4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to back translate a protein alignment using the sequences present on the current alignment. <br />
 Current alignment should contain the original sequences which were translated to protein sequences, <b> without any gap.</b>  <a href="#a57cb9cad92f31ec56982a4191d2cbce4">More...</a><br /></td></tr>
<tr class="separator:a57cb9cad92f31ec56982a4191d2cbce4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a12f7c76645e480ee17e7c1093d906ef8"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a12f7c76645e480ee17e7c1093d906ef8">checkCorrespondence</a> (std::string *names, int *lenghts, int totalInputSequences, int multiple)</td></tr>
<tr class="memdesc:a12f7c76645e480ee17e7c1093d906ef8"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to check CDS file.<br />
.  <a href="#a12f7c76645e480ee17e7c1093d906ef8">More...</a><br /></td></tr>
<tr class="separator:a12f7c76645e480ee17e7c1093d906ef8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac01973eecee182c962dd8cfe450b407e"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ac01973eecee182c962dd8cfe450b407e">calculateColIdentity</a> (float *columnIdentity)</td></tr>
<tr class="memdesc:ac01973eecee182c962dd8cfe450b407e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method that calculates the columns identity value.<br />
This is, the frequency of the most present element in the column, being residue, indetermination and gap allowed.  <a href="#ac01973eecee182c962dd8cfe450b407e">More...</a><br /></td></tr>
<tr class="separator:ac01973eecee182c962dd8cfe450b407e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a810e35f810586ae8e933b432a11a628d"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a810e35f810586ae8e933b432a11a628d">setKeepSequencesFlag</a> (bool newFlagValue)</td></tr>
<tr class="memdesc:a810e35f810586ae8e933b432a11a628d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Keep Sequences setter.  <a href="#a810e35f810586ae8e933b432a11a628d">More...</a><br /></td></tr>
<tr class="separator:a810e35f810586ae8e933b432a11a628d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aecf622bc94ed1fe6fe83e57cc849414e"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#aecf622bc94ed1fe6fe83e57cc849414e">printAlignmentInfo</a> (std::ostream &amp;output)</td></tr>
<tr class="memdesc:aecf622bc94ed1fe6fe83e57cc849414e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Print information about sequences number, average sequence length, maximum and minimum sequences length, etc.  <a href="#aecf622bc94ed1fe6fe83e57cc849414e">More...</a><br /></td></tr>
<tr class="separator:aecf622bc94ed1fe6fe83e57cc849414e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3fd2333ea7098410cb271b795f482822"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a3fd2333ea7098410cb271b795f482822">prepareCodingSequence</a> (bool splitByStopCodon, bool ignoreStopCodon, <a class="el" href="class_alignment.html">Alignment</a> *proteinAlignment)</td></tr>
<tr class="memdesc:a3fd2333ea7098410cb271b795f482822"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to check if the CDS file is correct.<br />
Based on nature of residues: DNA/RNA (Most of the residues)<br />
There is no gaps on the whole dataset.<br />
Each sequence is a multiple of 3.<br />
<br />
It will also remove or split stop codons depending on the flags passed.  <a href="#a3fd2333ea7098410cb271b795f482822">More...</a><br /></td></tr>
<tr class="separator:a3fd2333ea7098410cb271b795f482822"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3754dfadea173918f5e85b9eb4c3ad8a"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a3754dfadea173918f5e85b9eb4c3ad8a">alignmentSummaryHTML</a> (const <a class="el" href="class_alignment.html">Alignment</a> &amp;trimmedAlig, const char *const destFile)</td></tr>
<tr class="memdesc:a3754dfadea173918f5e85b9eb4c3ad8a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to report the trimming results in HTML.<br />
Outputs an HTML file that shows visually what has been done to the alignment (removed sequences or residues) and the stats used to do so.  <a href="#a3754dfadea173918f5e85b9eb4c3ad8a">More...</a><br /></td></tr>
<tr class="separator:a3754dfadea173918f5e85b9eb4c3ad8a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a724218d5e7d168974d9abcafcde632a1"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a724218d5e7d168974d9abcafcde632a1">statSVG</a> (const char *const destFile)</td></tr>
<tr class="separator:a724218d5e7d168974d9abcafcde632a1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4196c026e0f482eaccf9976f82f72e3"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ac4196c026e0f482eaccf9976f82f72e3">alignmentSummarySVG</a> (<a class="el" href="class_alignment.html">Alignment</a> &amp;trimmedAlig, const char *destFile, int blocks)</td></tr>
<tr class="memdesc:ac4196c026e0f482eaccf9976f82f72e3"><td class="mdescLeft">&#160;</td><td class="mdescRight">Method to report the trimming results in SVG.<br />
Outputs a SVG file that shows visually what has been done to the alignment (removed sequences or residues) and the stats used to do so.  <a href="#ac4196c026e0f482eaccf9976f82f72e3">More...</a><br /></td></tr>
<tr class="separator:ac4196c026e0f482eaccf9976f82f72e3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0b32dd76a1a98278068a7b65797ae6b9"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a0b32dd76a1a98278068a7b65797ae6b9">updateSequencesAndResiduesNums</a> (bool countSequences=true, bool countResidues=true)</td></tr>
<tr class="memdesc:a0b32dd76a1a98278068a7b65797ae6b9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Updates the sequence number and residue number based on saveResidues and saveSequences.  <a href="#a0b32dd76a1a98278068a7b65797ae6b9">More...</a><br /></td></tr>
<tr class="separator:a0b32dd76a1a98278068a7b65797ae6b9"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:ad92e860b13002fd1cd9fcc4c65818024"><td class="memItemLeft" align="right" valign="top"><a class="el" href="class_cleaner.html">Cleaner</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ad92e860b13002fd1cd9fcc4c65818024">Cleaning</a> = nullptr</td></tr>
<tr class="memdesc:ad92e860b13002fd1cd9fcc4c65818024"><td class="mdescLeft">&#160;</td><td class="mdescRight">Trimming submodule.<br />
It contains methods and variables related to trimming.  <a href="#ad92e860b13002fd1cd9fcc4c65818024">More...</a><br /></td></tr>
<tr class="separator:ad92e860b13002fd1cd9fcc4c65818024"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a26d027409ce208953560d3e5916ad616"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classstatistics_1_1_manager.html">statistics::Manager</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a26d027409ce208953560d3e5916ad616">Statistics</a> = nullptr</td></tr>
<tr class="memdesc:a26d027409ce208953560d3e5916ad616"><td class="mdescLeft">&#160;</td><td class="mdescRight">Statistics submodule.<br />
It contains methods and variables related to statistics calculation and reporting.  <a href="#a26d027409ce208953560d3e5916ad616">More...</a><br /></td></tr>
<tr class="separator:a26d027409ce208953560d3e5916ad616"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a032a37b570890ebfb3a925c329c36b43"><td class="memItemLeft" align="right" valign="top"><a class="el" href="class_alignment_1_1sequences_matrix.html">sequencesMatrix</a> *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a032a37b570890ebfb3a925c329c36b43">SequencesMatrix</a> = nullptr</td></tr>
<tr class="memdesc:a032a37b570890ebfb3a925c329c36b43"><td class="mdescLeft">&#160;</td><td class="mdescRight">SequencesMatrix submodule<br />
 This helper module stores the sequences without gaps. Useful when comparing MSA.  <a href="#a032a37b570890ebfb3a925c329c36b43">More...</a><br /></td></tr>
<tr class="separator:a032a37b570890ebfb3a925c329c36b43"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5373ff2b4fbeea7875a4613758276518"><td class="memItemLeft" align="right" valign="top">int *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518">SeqRef</a> = nullptr</td></tr>
<tr class="memdesc:a5373ff2b4fbeea7875a4613758276518"><td class="mdescLeft">&#160;</td><td class="mdescRight">Pointer to keep a count of how many alignments depend on the same shared-sequences.<br />
 Increased on each trim step, and automatically reduced on each alignment destroy.<br />
 If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information.  <a href="#a5373ff2b4fbeea7875a4613758276518">More...</a><br /></td></tr>
<tr class="separator:a5373ff2b4fbeea7875a4613758276518"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7edb0bea8267e7b9ac1b1171e14f9436"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a7edb0bea8267e7b9ac1b1171e14f9436">originalNumberOfSequences</a> = 0</td></tr>
<tr class="memdesc:a7edb0bea8267e7b9ac1b1171e14f9436"><td class="mdescLeft">&#160;</td><td class="mdescRight">Number of sequences the alignment had when it was loaded.  <a href="#a7edb0bea8267e7b9ac1b1171e14f9436">More...</a><br /></td></tr>
<tr class="separator:a7edb0bea8267e7b9ac1b1171e14f9436"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad236d6ee5f8414dead5b9dabca27c291"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ad236d6ee5f8414dead5b9dabca27c291">numberOfSequences</a> = 0</td></tr>
<tr class="memdesc:ad236d6ee5f8414dead5b9dabca27c291"><td class="mdescLeft">&#160;</td><td class="mdescRight">Number of sequences present on the alignment.  <a href="#ad236d6ee5f8414dead5b9dabca27c291">More...</a><br /></td></tr>
<tr class="separator:ad236d6ee5f8414dead5b9dabca27c291"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acb820d1138c4f0eda3d7c7d2ffe8fe66"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#acb820d1138c4f0eda3d7c7d2ffe8fe66">originalNumberOfResidues</a> = 0</td></tr>
<tr class="memdesc:acb820d1138c4f0eda3d7c7d2ffe8fe66"><td class="mdescLeft">&#160;</td><td class="mdescRight">Number of residues the alignment had when it was loaded.  <a href="#acb820d1138c4f0eda3d7c7d2ffe8fe66">More...</a><br /></td></tr>
<tr class="separator:acb820d1138c4f0eda3d7c7d2ffe8fe66"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5754ea57c16c29a35a36243fa3746b70"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a5754ea57c16c29a35a36243fa3746b70">numberOfResidues</a> = 0</td></tr>
<tr class="memdesc:a5754ea57c16c29a35a36243fa3746b70"><td class="mdescLeft">&#160;</td><td class="mdescRight">Number of residues present on the alignment if it is aligned.  <a href="#a5754ea57c16c29a35a36243fa3746b70">More...</a><br /></td></tr>
<tr class="separator:a5754ea57c16c29a35a36243fa3746b70"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae55f8a04d67451b23615467d24823daa"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ae55f8a04d67451b23615467d24823daa">isAligned</a> = false</td></tr>
<tr class="memdesc:ae55f8a04d67451b23615467d24823daa"><td class="mdescLeft">&#160;</td><td class="mdescRight">Flag that indicates if all sequences on the alignment have the same length (Including gaps).  <a href="#ae55f8a04d67451b23615467d24823daa">More...</a><br /></td></tr>
<tr class="separator:ae55f8a04d67451b23615467d24823daa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab0cafdefe306c19917a8c58abd2f8918"><td class="memItemLeft" align="right" valign="top">std::string *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ab0cafdefe306c19917a8c58abd2f8918">sequences</a> = nullptr</td></tr>
<tr class="memdesc:ab0cafdefe306c19917a8c58abd2f8918"><td class="mdescLeft">&#160;</td><td class="mdescRight">Vector containing the sequences.  <a href="#ab0cafdefe306c19917a8c58abd2f8918">More...</a><br /></td></tr>
<tr class="separator:ab0cafdefe306c19917a8c58abd2f8918"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afae240350928bf95b01456cbeb631767"><td class="memItemLeft" align="right" valign="top">std::string *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#afae240350928bf95b01456cbeb631767">seqsName</a> = nullptr</td></tr>
<tr class="memdesc:afae240350928bf95b01456cbeb631767"><td class="mdescLeft">&#160;</td><td class="mdescRight">String vector containing the sequences names.  <a href="#afae240350928bf95b01456cbeb631767">More...</a><br /></td></tr>
<tr class="separator:afae240350928bf95b01456cbeb631767"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abe93ddf27b0ef2c3cb7f4d0232a4d675"><td class="memItemLeft" align="right" valign="top">std::string *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#abe93ddf27b0ef2c3cb7f4d0232a4d675">seqsInfo</a> = nullptr</td></tr>
<tr class="memdesc:abe93ddf27b0ef2c3cb7f4d0232a4d675"><td class="mdescLeft">&#160;</td><td class="mdescRight">Vector containing extra information about each sequence.  <a href="#abe93ddf27b0ef2c3cb7f4d0232a4d675">More...</a><br /></td></tr>
<tr class="separator:abe93ddf27b0ef2c3cb7f4d0232a4d675"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa0ab035fcc63d3245a93556b1280592f"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#aa0ab035fcc63d3245a93556b1280592f">filename</a></td></tr>
<tr class="memdesc:aa0ab035fcc63d3245a93556b1280592f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Filename where this alignment was loaded from.  <a href="#aa0ab035fcc63d3245a93556b1280592f">More...</a><br /></td></tr>
<tr class="separator:aa0ab035fcc63d3245a93556b1280592f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4e411b6b93a4f9ad2c99aab2fbc20302"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a4e411b6b93a4f9ad2c99aab2fbc20302">alignmentInfo</a></td></tr>
<tr class="memdesc:a4e411b6b93a4f9ad2c99aab2fbc20302"><td class="mdescLeft">&#160;</td><td class="mdescRight">String containing information of the alignment as a whole.  <a href="#a4e411b6b93a4f9ad2c99aab2fbc20302">More...</a><br /></td></tr>
<tr class="separator:a4e411b6b93a4f9ad2c99aab2fbc20302"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aeb3c72eeb5d2954ad68369d48f8c16e6"><td class="memItemLeft" align="right" valign="top">float **&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#aeb3c72eeb5d2954ad68369d48f8c16e6">identities</a> = nullptr</td></tr>
<tr class="memdesc:aeb3c72eeb5d2954ad68369d48f8c16e6"><td class="mdescLeft">&#160;</td><td class="mdescRight">2D Matrix that represents how much each sequence resembles other sequences in the same MSA  <a href="#aeb3c72eeb5d2954ad68369d48f8c16e6">More...</a><br /></td></tr>
<tr class="separator:aeb3c72eeb5d2954ad68369d48f8c16e6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1b63fac4259b20561bf46369d8908d82"><td class="memItemLeft" align="right" valign="top">float **&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a1b63fac4259b20561bf46369d8908d82">overlaps</a> = nullptr</td></tr>
<tr class="memdesc:a1b63fac4259b20561bf46369d8908d82"><td class="mdescLeft">&#160;</td><td class="mdescRight">Matrix that stores the overlap between sequences of an alignment. See <a class="el" href="class_alignment.html#ab4d73aec4e1ec82bee55321791a1c6dc" title="Calculates overlap between sequences. ">Alignment::calculateSeqOverlap</a>.  <a href="#a1b63fac4259b20561bf46369d8908d82">More...</a><br /></td></tr>
<tr class="separator:a1b63fac4259b20561bf46369d8908d82"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad8384252e067c742dc5d249b34925891"><td class="memItemLeft" align="right" valign="top">int *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ad8384252e067c742dc5d249b34925891">saveResidues</a> = nullptr</td></tr>
<tr class="memdesc:ad8384252e067c742dc5d249b34925891"><td class="mdescLeft">&#160;</td><td class="mdescRight">Vector containing which residues will be kept/removed.<br />
-1 Indicates a 'removed/toRemove'. Otherwise residue/column will be kept.  <a href="#ad8384252e067c742dc5d249b34925891">More...</a><br /></td></tr>
<tr class="separator:ad8384252e067c742dc5d249b34925891"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a415ed54c53c4775263659e6d8a25e998"><td class="memItemLeft" align="right" valign="top">int *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#a415ed54c53c4775263659e6d8a25e998">saveSequences</a> = nullptr</td></tr>
<tr class="memdesc:a415ed54c53c4775263659e6d8a25e998"><td class="mdescLeft">&#160;</td><td class="mdescRight">Vector containing which sequences will be kept/removed.<br />
-1 Indicates a 'removed/toRemove'. Otherwise sequence will be kept.  <a href="#a415ed54c53c4775263659e6d8a25e998">More...</a><br /></td></tr>
<tr class="separator:a415ed54c53c4775263659e6d8a25e998"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-attribs"></a>
Private Attributes</h2></td></tr>
<tr class="memitem:ad9df4d89e17aea55da3792967d4a430b"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_alignment.html#ad9df4d89e17aea55da3792967d4a430b">dataType</a> = <a class="el" href="defines_8h.html#a69d5409d6e7370d6bd3b4f931702e936a3391d45e4b0be480216715453e5d9ddf">SequenceTypes::NotDefined</a></td></tr>
<tr class="memdesc:ad9df4d89e17aea55da3792967d4a430b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Flag variable that represents the type of the alignment.  <a href="#ad9df4d89e17aea55da3792967d4a430b">More...</a><br /></td></tr>
<tr class="separator:ad9df4d89e17aea55da3792967d4a430b"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Class containing an alignment<br />
 This class stores the alignment sequences with it's names, residues and extra information.<br />
 It contains multiple methods regarding the sequences.<br />
 It also contains submodules that provide methods for<b> Calculating statistics</b>, <b> Trim the alignment</b> and<b> Printing sequences information</b>. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00049">49</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a id="a091e21f0cfb741cb5cdc3535c6cc287f"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a091e21f0cfb741cb5cdc3535c6cc287f">&#9670;&nbsp;</a></span>Alignment() <span class="overload">[1/2]</span></h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::Alignment </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Constructor. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">46</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_cleaner_8cpp_source.html#l01633">Cleaner::Cleaner()</a>, <a class="el" href="_alignment_8h_source.html#l00058">Cleaning</a>, <a class="el" href="_alignment_8h_source.html#l00051">dataType</a>, <a class="el" href="_alignment_8h_source.html#l00089">identities</a>, <a class="el" href="_alignment_8h_source.html#l00076">isAligned</a>, <a class="el" href="_manager_8cpp_source.html#l00184">statistics::Manager::Manager()</a>, <a class="el" href="defines_8h_source.html#l00076">NotDefined</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00066">SeqRef</a>, <a class="el" href="_alignment_8h_source.html#l00082">seqsInfo</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, <a class="el" href="_alignment_8h_source.html#l00062">SequencesMatrix</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, and <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>.</p>
<div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a091e21f0cfb741cb5cdc3535c6cc287f_cgraph.svg" width="339" height="104"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe></div>
</div>
<div id="dynsection-1" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-1-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-1-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-1-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a091e21f0cfb741cb5cdc3535c6cc287f_icgraph.svg" width="1207" height="118"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="aeeee94970a445f32aa79646c249befef"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aeeee94970a445f32aa79646c249befef">&#9670;&nbsp;</a></span>Alignment() <span class="overload">[2/2]</span></h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::Alignment </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="class_alignment.html">Alignment</a> &amp;&#160;</td>
          <td class="paramname"><em>originalAlignment</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Copy Constructor. <br />
 The copy constructor does not make a copy of all structures present on the original alignment.<br />
 Instead, all alignments that share a common origin, share some information, and also, a counter (<a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a>) of how many alignments share the same information.<br />
 This allows us to keep the information on memory while any alignment is using it.<br />
 When an alignment is destroyed, the counter is substracted by 1, and, if it is 0 after the substraction, the shared data is destroyed. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00087">87</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00086">alignmentInfo</a>, <a class="el" href="_cleaner_8cpp_source.html#l01648">Cleaner::Cleaner()</a>, <a class="el" href="_alignment_8h_source.html#l00058">Cleaning</a>, <a class="el" href="_alignment_8h_source.html#l00051">dataType</a>, <a class="el" href="_alignment_8h_source.html#l00084">filename</a>, <a class="el" href="_alignment_8h_source.html#l00089">identities</a>, <a class="el" href="_alignment_8h_source.html#l00076">isAligned</a>, <a class="el" href="_manager_8cpp_source.html#l00194">statistics::Manager::Manager()</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00066">SeqRef</a>, <a class="el" href="_alignment_8h_source.html#l00082">seqsInfo</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, <a class="el" href="_alignment_8h_source.html#l00062">SequencesMatrix</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

<p class="reference">Referenced by <a class="el" href="_cleaner_8cpp_source.html#l00231">Cleaner::cleanByCutValueFallBehind()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00090">Cleaner::cleanByCutValueOverpass()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00357">Cleaner::cleanByCutValueOverpassOrEquals()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00714">Cleaner::cleanOverlapSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00513">Cleaner::cleanStrict()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01211">Cleaner::getClustering()</a>, <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01240">Cleaner::removeColumns()</a>, and <a class="el" href="_cleaner_8cpp_source.html#l01262">Cleaner::removeSequences()</a>.</p>
<div id="dynsection-2" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-2-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-2-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-2-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_aeeee94970a445f32aa79646c249befef_cgraph.svg" width="543" height="170"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-3" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-3-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-3-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-3-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_aeeee94970a445f32aa79646c249befef_icgraph.svg" width="1704" height="611"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a8787573a5060136c2a51f40cf229e356"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a8787573a5060136c2a51f40cf229e356">&#9670;&nbsp;</a></span>~Alignment()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::~Alignment </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Destructor. The destructor takes care of removing the non-shared information of each alignment when is destroyed.<br />
 It also substracts 1 from the shared counter <a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a>.<br />
 When <a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a> arrives to 0, the shared data is destroyed, including <a class="el" href="class_alignment.html#a5373ff2b4fbeea7875a4613758276518" title="Pointer to keep a count of how many alignments depend on the same shared-sequences.  Increased on each trim step, and automatically reduced on each alignment destroy.  If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. ">Alignment::SeqRef</a>. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">134</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00058">Cleaning</a>, <a class="el" href="_alignment_8h_source.html#l00089">identities</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00066">SeqRef</a>, <a class="el" href="_alignment_8h_source.html#l00082">seqsInfo</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, <a class="el" href="_alignment_8h_source.html#l00062">SequencesMatrix</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a id="a3754dfadea173918f5e85b9eb4c3ad8a"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a3754dfadea173918f5e85b9eb4c3ad8a">&#9670;&nbsp;</a></span>alignmentSummaryHTML()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::alignmentSummaryHTML </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="class_alignment.html">Alignment</a> &amp;&#160;</td>
          <td class="paramname"><em>trimmedAlig</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *const&#160;</td>
          <td class="paramname"><em>destFile</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to report the trimming results in HTML.<br />
Outputs an HTML file that shows visually what has been done to the alignment (removed sequences or residues) and the stats used to do so. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">trimmedAlig</td><td>Trimmed alignment to compare with. </td></tr>
    <tr><td class="paramname">destFile</td><td>Filename where to save the results. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><b>True</b> if correct. <br />
 <b>False</b> if file couldn't be open or alignment is not aligned. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">1167</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_manager_8h_source.html#l00067">statistics::Manager::consistency</a>, <a class="el" href="reportsystem_8cpp_source.html#l00033">debug</a>, <a class="el" href="utils_8cpp_source.html#l00675">utils::determineColor()</a>, <a class="el" href="_alignment_8h_source.html#l00084">filename</a>, <a class="el" href="_manager_8h_source.html#l00059">statistics::Manager::gaps</a>, <a class="el" href="_gaps_8cpp_source.html#l00163">statistics::Gaps::getGapsWindow()</a>, <a class="el" href="_similarity_8cpp_source.html#l00338">statistics::Similarity::getMdkWindowedVector()</a>, <a class="el" href="_consistency_8cpp_source.html#l00588">statistics::Consistency::getValues()</a>, <a class="el" href="_alignment_8h_source.html#l00076">isAligned</a>, <a class="el" href="utils_8cpp_source.html#l00080">utils::max()</a>, <a class="el" href="reportsystem_8h_source.html#l00134">NotAligned</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="reportsystem_8cpp_source.html#l00067">reporting::reportManager::report()</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, <a class="el" href="_manager_8h_source.html#l00063">statistics::Manager::similarity</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01884">trimAlManager::output_reports()</a>.</p>
<div id="dynsection-4" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-4-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-4-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-4-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a3754dfadea173918f5e85b9eb4c3ad8a_cgraph.svg" width="1714" height="638"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-5" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-5-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-5-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-5-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a3754dfadea173918f5e85b9eb4c3ad8a_icgraph.svg" width="987" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="ac4196c026e0f482eaccf9976f82f72e3"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ac4196c026e0f482eaccf9976f82f72e3">&#9670;&nbsp;</a></span>alignmentSummarySVG()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::alignmentSummarySVG </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="class_alignment.html">Alignment</a> &amp;&#160;</td>
          <td class="paramname"><em>trimmedAlig</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>destFile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>blocks</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to report the trimming results in SVG.<br />
Outputs a SVG file that shows visually what has been done to the alignment (removed sequences or residues) and the stats used to do so. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">trimmedAlig</td><td>Trimmed alignment obtained from the object calling this function. </td></tr>
    <tr><td class="paramname">destFile</td><td>Filename where to save the results. </td></tr>
    <tr><td class="paramname">blocks</td><td>Size in residues to report. 120 by default. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><b>True</b> if everything went ok. <br />
 <b>False</b> if file couldn't be open or alignment is not aligned. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">1708</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_manager_8h_source.html#l00067">statistics::Manager::consistency</a>, <a class="el" href="reportsystem_8cpp_source.html#l00033">debug</a>, <a class="el" href="utils_8cpp_source.html#l00675">utils::determineColor()</a>, <a class="el" href="_alignment_8h_source.html#l00084">filename</a>, <a class="el" href="_manager_8h_source.html#l00059">statistics::Manager::gaps</a>, <a class="el" href="_gaps_8cpp_source.html#l00163">statistics::Gaps::getGapsWindow()</a>, <a class="el" href="_similarity_8cpp_source.html#l00338">statistics::Similarity::getMdkWindowedVector()</a>, <a class="el" href="_consistency_8cpp_source.html#l00588">statistics::Consistency::getValues()</a>, <a class="el" href="_alignment_8h_source.html#l00076">isAligned</a>, <a class="el" href="utils_8cpp_source.html#l00080">utils::max()</a>, <a class="el" href="reportsystem_8h_source.html#l00134">NotAligned</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="reportsystem_8cpp_source.html#l00067">reporting::reportManager::report()</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, <a class="el" href="_manager_8h_source.html#l00063">statistics::Manager::similarity</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01884">trimAlManager::output_reports()</a>.</p>
<div id="dynsection-6" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-6-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-6-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-6-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ac4196c026e0f482eaccf9976f82f72e3_cgraph.svg" width="1706" height="638"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-7" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-7-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-7-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-7-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ac4196c026e0f482eaccf9976f82f72e3_icgraph.svg" width="979" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="ac01973eecee182c962dd8cfe450b407e"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ac01973eecee182c962dd8cfe450b407e">&#9670;&nbsp;</a></span>calculateColIdentity()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::calculateColIdentity </td>
          <td>(</td>
          <td class="paramtype">float *&#160;</td>
          <td class="paramname"><em>columnIdentity</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method that calculates the columns identity value.<br />
This is, the frequency of the most present element in the column, being residue, indetermination and gap allowed. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramdir">[out]</td><td class="paramname">columnIdentity</td><td>Vector to fill with identities for each column. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01074">1074</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="defines_8h_source.html#l00088">AA</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>
<div id="dynsection-8" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-8-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-8-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-8-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ac01973eecee182c962dd8cfe450b407e_cgraph.svg" width="827" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="ab4d73aec4e1ec82bee55321791a1c6dc"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ab4d73aec4e1ec82bee55321791a1c6dc">&#9670;&nbsp;</a></span>calculateSeqOverlap()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::calculateSeqOverlap </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Calculates overlap between sequences. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">378</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="defines_8h_source.html#l00088">AA</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00092">overlaps</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01001">printSeqOverlap()</a>.</p>
<div id="dynsection-9" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-9-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-9-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-9-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ab4d73aec4e1ec82bee55321791a1c6dc_cgraph.svg" width="832" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-10" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-10-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-10-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-10-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ab4d73aec4e1ec82bee55321791a1c6dc_icgraph.svg" width="1176" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a12f7c76645e480ee17e7c1093d906ef8"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a12f7c76645e480ee17e7c1093d906ef8">&#9670;&nbsp;</a></span>checkCorrespondence()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::checkCorrespondence </td>
          <td>(</td>
          <td class="paramtype">std::string *&#160;</td>
          <td class="paramname"><em>names</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int *&#160;</td>
          <td class="paramname"><em>lenghts</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>totalInputSequences</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>multiple</em> = <code>1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to check CDS file.<br />
. </p>
<ul>
<li>It checks if sequences of input alignment are all present on the CDS file.</li>
<li>If nucleotide sequence is larger than protein sequence length * 3, warns about it and cuts the nucleotide sequence</li>
<li>If sequence has indetermination symbols, it warns about it.</li>
<li>If nucleotide sequence is smaller than protein sequence * 3, it adds some 'N' at the end of the nucleotide sequence. <dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">names</td><td>Vector containing the names to check. </td></tr>
    <tr><td class="paramname">lenghts</td><td>Vector containing the length of each sequence. </td></tr>
    <tr><td class="paramname">totalInputSequences</td><td>Number of sequences present. </td></tr>
    <tr><td class="paramname">multiple</td><td>Multiplies the length of each sequence by this number. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><b>True</b> if all went right.<br />
 <b>False</b> if there is a sequence in the alignment but not in the names vector. </dd></dl>
</li>
</ul>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00716">716</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="reportsystem_8cpp_source.html#l00033">debug</a>, <a class="el" href="reportsystem_8h_source.html#l00273">IncludingIndeterminationSymbols</a>, <a class="el" href="reportsystem_8h_source.html#l00275">LessNucleotidesThanExpected</a>, <a class="el" href="utils_8cpp_source.html#l00098">utils::min()</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="utils_8cpp_source.html#l00485">utils::removeCharacter()</a>, <a class="el" href="reportsystem_8cpp_source.html#l00120">reporting::reportManager::report()</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="reportsystem_8h_source.html#l00178">SequenceNotPresentInCDS</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, and <a class="el" href="reportsystem_8h_source.html#l00271">SequenceWillBeCut</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01600">trimAlManager::check_backtranslation_infile_names_correspondence()</a>.</p>
<div id="dynsection-11" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-11-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-11-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-11-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a12f7c76645e480ee17e7c1093d906ef8_cgraph.svg" width="440" height="155"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-12" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-12-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-12-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-12-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a12f7c76645e480ee17e7c1093d906ef8_icgraph.svg" width="1011" height="67"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a73e7500f4334361430be77f8d944be7f"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a73e7500f4334361430be77f8d944be7f">&#9670;&nbsp;</a></span>fillMatrices()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::fillMatrices </td>
          <td>(</td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>aligned</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>checkInvalidChars</em> = <code>true</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to initialize matrices and check if the sequences have been correctly loaded and are free of errors<br />
It checks if sequences contain unknown characters, sets the isAligned flag and initializes 'saveResidues' and 'saveSequences', depending on the sizes of the sequences (whether or not they have the same length). </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">aligned</td><td>Flag to make the method check if the alignment is aligned or not. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section note"><dt>Note</dt><dd>Even with the aligned flag set to false, if the alignment is aligned it will initialize the variables 'saveResidues' and 'saveSequences', allowing the alignment to be trimmed. </dd></dl>
<dl class="section return"><dt>Returns</dt><dd><b>True</b> if the alignment information was ok. <br />
 <b>False</b> if there was a problem.<br />
This could happen if the sequences contain any unknown character or if aligned flag is set up to true and the sequences have variable sizes. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">794</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="reportsystem_8cpp_source.html#l00033">debug</a>, <a class="el" href="_alignment_8h_source.html#l00084">filename</a>, <a class="el" href="_alignment_8h_source.html#l00076">isAligned</a>, <a class="el" href="reportsystem_8h_source.html#l00134">NotAligned</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="reportsystem_8cpp_source.html#l00067">reporting::reportManager::report()</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>, <a class="el" href="reportsystem_8h_source.html#l00182">SequencesNotSameSize</a>, and <a class="el" href="reportsystem_8h_source.html#l00180">UnknownCharacter</a>.</p>
<div id="dynsection-13" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-13-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-13-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-13-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a73e7500f4334361430be77f8d944be7f_cgraph.svg" width="372" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a3f8f2eafd37c9b5bc275b9889a2de96a"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a3f8f2eafd37c9b5bc275b9889a2de96a">&#9670;&nbsp;</a></span>getAlignmentType()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getAlignmentType </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p><a class="el" href="class_alignment.html" title="Class containing an alignment  This class stores the alignment sequences with it&#39;s names...">Alignment</a> type getter. <br />
 See <a class="el" href="defines_8h.html#a69d5409d6e7370d6bd3b4f931702e936">SequenceTypes</a>. </p>
<dl class="section return"><dt>Returns</dt><dd>Int representing the alignment type.<br />
</dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">469</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="utils_8cpp_source.html#l00499">utils::checkAlignmentType()</a>, <a class="el" href="_alignment_8h_source.html#l00051">dataType</a>, <a class="el" href="defines_8h_source.html#l00076">NotDefined</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01074">calculateColIdentity()</a>, <a class="el" href="_similarity_8cpp_source.html#l00097">statistics::Similarity::calculateMatrixIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">calculateSeqOverlap()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00855">Cleaner::calculateSpuriousVector()</a>, <a class="el" href="_similarity_8cpp_source.html#l00157">statistics::Similarity::calculateVectors()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01564">trimAlManager::check_coding_sequences_type()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01963">trimAlManager::create_or_use_similarity_matrix()</a>, <a class="el" href="read_al_main_8cpp_source.html#l00359">main()</a>, <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">prepareCodingSequence()</a>.</p>
<div id="dynsection-14" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-14-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-14-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-14-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a3f8f2eafd37c9b5bc275b9889a2de96a_cgraph.svg" width="576" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-15" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-15-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-15-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-15-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a3f8f2eafd37c9b5bc275b9889a2de96a_icgraph.svg" width="2890" height="1083"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a5b534495c7c6fac732b290359f3b8992"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a5b534495c7c6fac732b290359f3b8992">&#9670;&nbsp;</a></span>getNumAminos()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getNumAminos </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Residues number getter. It counts gaps as residue. </p>
<dl class="section return"><dt>Returns</dt><dd>Number of residues in the alignment. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00252">252</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01526">trimAlManager::check_block_size()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02145">trimAlManager::CleanResiduesNonAuto()</a>, <a class="el" href="_consistency_8cpp_source.html#l00241">statistics::Consistency::compareAndChoose()</a>, <a class="el" href="_consistency_8cpp_source.html#l00416">statistics::Consistency::forceComparison()</a>, and <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>.</p>
<div id="dynsection-16" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-16-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-16-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-16-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a5b534495c7c6fac732b290359f3b8992_icgraph.svg" width="1291" height="300"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="af9c7aec9082ef16573d7fe37a93a638a"></a>
<h2 class="memtitle"><span class="permalink"><a href="#af9c7aec9082ef16573d7fe37a93a638a">&#9670;&nbsp;</a></span>getNumSpecies()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getNumSpecies </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Number of sequences getter. </p>
<dl class="section return"><dt>Returns</dt><dd>Number of sequences in the alignment. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00245">245</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01600">trimAlManager::check_backtranslation_infile_names_correspondence()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01175">trimAlManager::check_clusters_incompatibilities()</a>, <a class="el" href="_consistency_8cpp_source.html#l00241">statistics::Consistency::compareAndChoose()</a>, <a class="el" href="_consistency_8cpp_source.html#l00416">statistics::Consistency::forceComparison()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">prepareCodingSequence()</a>.</p>
<div id="dynsection-17" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-17-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-17-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-17-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_af9c7aec9082ef16573d7fe37a93a638a_icgraph.svg" width="1315" height="316"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="ab24c9f56fb953e3e01eec2c850c6aee5"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ab24c9f56fb953e3e01eec2c850c6aee5">&#9670;&nbsp;</a></span>getSequenceNameOrder()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getSequenceNameOrder </td>
          <td>(</td>
          <td class="paramtype">std::string *&#160;</td>
          <td class="paramname"><em>names</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int *&#160;</td>
          <td class="paramname"><em>orderVector</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to map two sets of sequences, own and external.<br />
The method accepts a vector of names, and test them against the alignment own sequences' names.<br />
If the sequence is present, its index will be stored in orderVector.<br />
. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramdir"></td><td class="paramname">names</td><td>Vector of sequences names to map. </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">orderVector</td><td>Vector of orders to fill. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd><b>True</b> if both sets have the same sequences.<br />
 <b>False</b> otherwise. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00445">445</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, and <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>.</p>

<p class="reference">Referenced by <a class="el" href="_consistency_8cpp_source.html#l00241">statistics::Consistency::compareAndChoose()</a>, and <a class="el" href="_consistency_8cpp_source.html#l00416">statistics::Consistency::forceComparison()</a>.</p>
<div id="dynsection-18" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-18-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-18-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-18-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ab24c9f56fb953e3e01eec2c850c6aee5_icgraph.svg" width="1251" height="118"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="af5deab9f747b96c43a1004c2e1170095"></a>
<h2 class="memtitle"><span class="permalink"><a href="#af5deab9f747b96c43a1004c2e1170095">&#9670;&nbsp;</a></span>getSequences() <span class="overload">[1/3]</span></h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getSequences </td>
          <td>(</td>
          <td class="paramtype">std::string *&#160;</td>
          <td class="paramname"><em>names</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Getter for the sequences names. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramdir">[out]</td><td class="paramname">names</td><td>Vector of sequences names to fill. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00416">416</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, and <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>.</p>

<p class="reference">Referenced by <a class="el" href="_consistency_8cpp_source.html#l00241">statistics::Consistency::compareAndChoose()</a>, and <a class="el" href="_consistency_8cpp_source.html#l00416">statistics::Consistency::forceComparison()</a>.</p>
<div id="dynsection-19" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-19-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-19-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-19-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_af5deab9f747b96c43a1004c2e1170095_icgraph.svg" width="1223" height="118"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="ae8b5f6fbc7ffb828c7eb51ae74e6888c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae8b5f6fbc7ffb828c7eb51ae74e6888c">&#9670;&nbsp;</a></span>getSequences() <span class="overload">[2/3]</span></h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getSequences </td>
          <td>(</td>
          <td class="paramtype">std::string *&#160;</td>
          <td class="paramname"><em>names</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int *&#160;</td>
          <td class="paramname"><em>lenghts</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Getter for the sequences names and its lenghts. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramdir">[out]</td><td class="paramname">names</td><td>Vector of sequences names to fill. </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">lenghts</td><td>Vector of lenghts to fill. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00424">424</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="utils_8cpp_source.html#l00485">utils::removeCharacter()</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01600">trimAlManager::check_backtranslation_infile_names_correspondence()</a>.</p>
<div id="dynsection-20" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-20-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-20-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-20-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ae8b5f6fbc7ffb828c7eb51ae74e6888c_cgraph.svg" width="384" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-21" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-21-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-21-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-21-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_ae8b5f6fbc7ffb828c7eb51ae74e6888c_icgraph.svg" width="963" height="67"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a4dcdeecf3fad5bb749f3f6051655e17a"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a4dcdeecf3fad5bb749f3f6051655e17a">&#9670;&nbsp;</a></span>getSequences() <span class="overload">[3/3]</span></h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getSequences </td>
          <td>(</td>
          <td class="paramtype">std::string *&#160;</td>
          <td class="paramname"><em>names</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::string *&#160;</td>
          <td class="paramname"><em>sequences</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int *&#160;</td>
          <td class="paramname"><em>lenghts</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Getter for the sequences, its names and lenghts. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramdir">[out]</td><td class="paramname">names</td><td>Vector of sequences names to fill. </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">sequences</td><td>Vector of sequences to fill. </td></tr>
    <tr><td class="paramdir">[out]</td><td class="paramname">lenghts</td><td>Vector of lenghts to fill. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00434">434</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="utils_8cpp_source.html#l00485">utils::removeCharacter()</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">prepareCodingSequence()</a>.</p>
<div id="dynsection-22" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-22-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-22-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-22-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a4dcdeecf3fad5bb749f3f6051655e17a_cgraph.svg" width="384" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-23" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-23-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-23-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-23-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a4dcdeecf3fad5bb749f3f6051655e17a_icgraph.svg" width="1224" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a57cb9cad92f31ec56982a4191d2cbce4"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a57cb9cad92f31ec56982a4191d2cbce4">&#9670;&nbsp;</a></span>getTranslationCDS()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::getTranslationCDS </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="class_alignment.html">Alignment</a> *&#160;</td>
          <td class="paramname"><em>proteinAlignment</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to back translate a protein alignment using the sequences present on the current alignment. <br />
 Current alignment should contain the original sequences which were translated to protein sequences, <b> without any gap.</b> </p>
<dl class="section note"><dt>Note</dt><dd>It's recommended to translate from dna to protein and then, align the protein sequences. </dd></dl>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">proteinAlignment</td><td>Protein <a class="el" href="class_alignment.html" title="Class containing an alignment  This class stores the alignment sequences with it&#39;s names...">Alignment</a> to use as origin. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>Pointer to new alignment. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">169</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>, <a class="el" href="_alignment_8h_source.html#l00082">seqsInfo</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l02023">trimAlManager::postprocess_alignment()</a>.</p>
<div id="dynsection-24" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-24-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-24-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-24-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a57cb9cad92f31ec56982a4191d2cbce4_cgraph.svg" width="586" height="104"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-25" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-25-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-25-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-25-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a57cb9cad92f31ec56982a4191d2cbce4_icgraph.svg" width="984" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a4fdfd8529dd7d554260377352a68a571"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a4fdfd8529dd7d554260377352a68a571">&#9670;&nbsp;</a></span>isFileAligned()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::isFileAligned </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>isAligned getter. </p>
<dl class="section return"><dt>Returns</dt><dd>Whether the alignment is aligned or not. </dd></dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00495">495</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00076">isAligned</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01535">trimAlManager::check_backtranslations()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01349">trimAlManager::check_file_aligned()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02000">trimAlManager::clean_alignment()</a>, <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00868">printAlignmentInfo()</a>.</p>
<div id="dynsection-26" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-26-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-26-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-26-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a4fdfd8529dd7d554260377352a68a571_icgraph.svg" width="1062" height="300"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a3fd2333ea7098410cb271b795f482822"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a3fd2333ea7098410cb271b795f482822">&#9670;&nbsp;</a></span>prepareCodingSequence()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::prepareCodingSequence </td>
          <td>(</td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>splitByStopCodon</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>ignoreStopCodon</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="class_alignment.html">Alignment</a> *&#160;</td>
          <td class="paramname"><em>proteinAlignment</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to check if the CDS file is correct.<br />
Based on nature of residues: DNA/RNA (Most of the residues)<br />
There is no gaps on the whole dataset.<br />
Each sequence is a multiple of 3.<br />
<br />
It will also remove or split stop codons depending on the flags passed. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">splitByStopCodon</td><td>Flag that informs the method to split sequences if it finds any stop codon. </td></tr>
    <tr><td class="paramname">ignoreStopCodon</td><td>Flag that informs the method to stop reading sequence if it finds any stop codon. </td></tr>
    <tr><td class="paramname">proteinAlignment</td><td><a class="el" href="class_alignment.html" title="Class containing an alignment  This class stores the alignment sequences with it&#39;s names...">Alignment</a> containing protein sequences which names contains all names in the current alignment. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">528</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="defines_8h_source.html#l00088">AA</a>, <a class="el" href="reportsystem_8h_source.html#l00170">CDScontainsProteinSequences</a>, <a class="el" href="reportsystem_8h_source.html#l00292">CuttingSequence</a>, <a class="el" href="reportsystem_8cpp_source.html#l00033">debug</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00245">getNumSpecies()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00434">getSequences()</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="reportsystem_8cpp_source.html#l00067">reporting::reportManager::report()</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, <a class="el" href="reportsystem_8h_source.html#l00172">SequenceContainsGap</a>, <a class="el" href="reportsystem_8h_source.html#l00176">SequenceHasStopCodon</a>, <a class="el" href="reportsystem_8h_source.html#l00174">SequenceNotMultipleOfThree</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01581">trimAlManager::check_and_prepare_coding_sequence()</a>.</p>
<div id="dynsection-27" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-27-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-27-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-27-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a3fd2333ea7098410cb271b795f482822_cgraph.svg" width="796" height="206"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-28" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-28-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-28-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-28-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a3fd2333ea7098410cb271b795f482822_icgraph.svg" width="1006" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="aecf622bc94ed1fe6fe83e57cc849414e"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aecf622bc94ed1fe6fe83e57cc849414e">&#9670;&nbsp;</a></span>printAlignmentInfo()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::printAlignmentInfo </td>
          <td>(</td>
          <td class="paramtype">std::ostream &amp;&#160;</td>
          <td class="paramname"><em>output</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Print information about sequences number, average sequence length, maximum and minimum sequences length, etc. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">output</td><td>Output stream. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00868">868</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00495">isFileAligned()</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>, and <a class="el" href="_alignment_8h_source.html#l00078">sequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="read_al_main_8cpp_source.html#l00359">main()</a>.</p>
<div id="dynsection-29" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-29-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-29-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-29-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_aecf622bc94ed1fe6fe83e57cc849414e_cgraph.svg" width="415" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-30" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-30-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-30-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-30-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_aecf622bc94ed1fe6fe83e57cc849414e_icgraph.svg" width="303" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="accfd82f91b9cdbe7a14ba85a1bada8cf"></a>
<h2 class="memtitle"><span class="permalink"><a href="#accfd82f91b9cdbe7a14ba85a1bada8cf">&#9670;&nbsp;</a></span>printSeqIdentity()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::printSeqIdentity </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Method to print different identity values computed from the alignment.<br />
In this method we assess the identity values matrix, as well as different average values. <br />
Moreover, the method computes which one is the most similar sequence. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00912">912</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_alignment_8h_source.html#l00058">Cleaning</a>, <a class="el" href="_alignment_8h_source.html#l00089">identities</a>, <a class="el" href="utils_8cpp_source.html#l00080">utils::max()</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, and <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01915">trimAlManager::print_statistics()</a>.</p>
<div id="dynsection-31" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-31-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-31-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-31-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_accfd82f91b9cdbe7a14ba85a1bada8cf_cgraph.svg" width="1044" height="88"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-32" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-32-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-32-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-32-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_accfd82f91b9cdbe7a14ba85a1bada8cf_icgraph.svg" width="918" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a57b5686aff7324890422aa4295bc62d1"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a57b5686aff7324890422aa4295bc62d1">&#9670;&nbsp;</a></span>printSeqOverlap()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::printSeqOverlap </td>
          <td>(</td>
          <td class="paramtype">void&#160;</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Prints the overlap between sequences. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01001">1001</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">calculateSeqOverlap()</a>, <a class="el" href="utils_8cpp_source.html#l00080">utils::max()</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00092">overlaps</a>, and <a class="el" href="_alignment_8h_source.html#l00080">seqsName</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01915">trimAlManager::print_statistics()</a>.</p>
<div id="dynsection-33" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-33-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-33-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-33-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a57b5686aff7324890422aa4295bc62d1_cgraph.svg" width="1062" height="88"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-34" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-34-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-34-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-34-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a57b5686aff7324890422aa4295bc62d1_icgraph.svg" width="920" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="acf3b1ef007f2903570bb2c80105f1876"></a>
<h2 class="memtitle"><span class="permalink"><a href="#acf3b1ef007f2903570bb2c80105f1876">&#9670;&nbsp;</a></span>setBlockSize()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::setBlockSize </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>blockSize</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>BlockSize Setter. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">blockSize</td><td>New value. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00267">267</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_cleaner_8h_source.html#l00067">Cleaner::blockSize</a>, and <a class="el" href="_alignment_8h_source.html#l00058">Cleaning</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01803">trimAlManager::innerPerform()</a>.</p>
<div id="dynsection-35" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-35-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-35-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-35-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_acf3b1ef007f2903570bb2c80105f1876_icgraph.svg" width="718" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a810e35f810586ae8e933b432a11a628d"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a810e35f810586ae8e933b432a11a628d">&#9670;&nbsp;</a></span>setKeepSequencesFlag()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::setKeepSequencesFlag </td>
          <td>(</td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>newFlagValue</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Keep Sequences setter. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">newFlagValue</td><td>New flag value </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00274">274</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00058">Cleaning</a>, and <a class="el" href="_cleaner_8h_source.html#l00063">Cleaner::keepSequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01803">trimAlManager::innerPerform()</a>.</p>
<div id="dynsection-36" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-36-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-36-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-36-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a810e35f810586ae8e933b432a11a628d_icgraph.svg" width="779" height="38"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a9f45aaea7bd507135890b1c8d37db930"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a9f45aaea7bd507135890b1c8d37db930">&#9670;&nbsp;</a></span>setWindowsSize()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::setWindowsSize </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>ghWindow</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>shWindow</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Windows setter. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">ghWindow</td><td>Half the Gap Window. </td></tr>
    <tr><td class="paramname">shWindow</td><td>Half the Similarity Window. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00259">259</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_manager_8h_source.html#l00077">statistics::Manager::ghWindow</a>, <a class="el" href="_manager_8h_source.html#l00081">statistics::Manager::shWindow</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l02226">trimAlManager::set_window_size()</a>.</p>
<div id="dynsection-37" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-37-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-37-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-37-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a9f45aaea7bd507135890b1c8d37db930_icgraph.svg" width="919" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a724218d5e7d168974d9abcafcde632a1"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a724218d5e7d168974d9abcafcde632a1">&#9670;&nbsp;</a></span>statSVG()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool Alignment::statSVG </td>
          <td>(</td>
          <td class="paramtype">const char *const&#160;</td>
          <td class="paramname"><em>destFile</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01431">1431</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_manager_8h_source.html#l00067">statistics::Manager::consistency</a>, <a class="el" href="_alignment_8h_source.html#l00084">filename</a>, <a class="el" href="_manager_8h_source.html#l00059">statistics::Manager::gaps</a>, <a class="el" href="_gaps_8cpp_source.html#l00163">statistics::Gaps::getGapsWindow()</a>, <a class="el" href="_similarity_8cpp_source.html#l00338">statistics::Similarity::getMdkWindowedVector()</a>, <a class="el" href="_consistency_8cpp_source.html#l00588">statistics::Consistency::getValues()</a>, <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="utils_8cpp_source.html#l00159">utils::quicksort()</a>, <a class="el" href="_manager_8h_source.html#l00063">statistics::Manager::similarity</a>, and <a class="el" href="_alignment_8h_source.html#l00060">Statistics</a>.</p>

<p class="reference">Referenced by <a class="el" href="trimal_manager_8cpp_source.html#l01958">trimAlManager::svg_stats_out()</a>.</p>
<div id="dynsection-38" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-38-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-38-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-38-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a724218d5e7d168974d9abcafcde632a1_cgraph.svg" width="1616" height="508"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>
<div id="dynsection-39" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-39-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-39-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-39-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a724218d5e7d168974d9abcafcde632a1_icgraph.svg" width="874" height="52"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<a id="a0b32dd76a1a98278068a7b65797ae6b9"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a0b32dd76a1a98278068a7b65797ae6b9">&#9670;&nbsp;</a></span>updateSequencesAndResiduesNums()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::updateSequencesAndResiduesNums </td>
          <td>(</td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>countSequences</em> = <code>true</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>countResidues</em> = <code>true</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Updates the sequence number and residue number based on saveResidues and saveSequences. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">countSequences</td><td>Whether to count sequences </td></tr>
    <tr><td class="paramname">countResidues</td><td>Whether to count residues </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">2347</a> of file <a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a>.</p>

<p class="reference">References <a class="el" href="_alignment_8h_source.html#l00074">numberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00070">numberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00072">originalNumberOfResidues</a>, <a class="el" href="_alignment_8h_source.html#l00068">originalNumberOfSequences</a>, <a class="el" href="_alignment_8h_source.html#l00095">saveResidues</a>, and <a class="el" href="_alignment_8h_source.html#l00096">saveSequences</a>.</p>

<p class="reference">Referenced by <a class="el" href="_cleaner_8cpp_source.html#l01240">Cleaner::removeColumns()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01283">Cleaner::removeOnlyTerminal()</a>, and <a class="el" href="_cleaner_8cpp_source.html#l01262">Cleaner::removeSequences()</a>.</p>
<div id="dynsection-40" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-40-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-40-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-40-content" class="dyncontent" style="display:none;">
<div class="center"><iframe scrolling="no" frameborder="0" src="class_alignment_a0b32dd76a1a98278068a7b65797ae6b9_icgraph.svg" width="1470" height="163"><p><b>This browser is not able to show SVG: try Firefox, Chrome, Safari, or Opera instead.</b></p></iframe>
</div>
</div>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a id="a4e411b6b93a4f9ad2c99aab2fbc20302"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a4e411b6b93a4f9ad2c99aab2fbc20302">&#9670;&nbsp;</a></span>alignmentInfo</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::alignmentInfo</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>String containing information of the alignment as a whole. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00086">86</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00087">Alignment()</a>.</p>

</div>
</div>
<a id="ad92e860b13002fd1cd9fcc4c65818024"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ad92e860b13002fd1cd9fcc4c65818024">&#9670;&nbsp;</a></span>Cleaning</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::Cleaning = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Trimming submodule.<br />
It contains methods and variables related to trimming. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00058">58</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01525">Cleaner::calculateRepresentativeSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00231">Cleaner::cleanByCutValueFallBehind()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00090">Cleaner::cleanByCutValueOverpass()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00357">Cleaner::cleanByCutValueOverpassOrEquals()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00714">Cleaner::cleanOverlapSeq()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02108">trimAlManager::CleanResiduesAuto()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02145">trimAlManager::CleanResiduesNonAuto()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02051">trimAlManager::CleanSequences()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00513">Cleaner::cleanStrict()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01803">trimAlManager::innerPerform()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02023">trimAlManager::postprocess_alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00912">printSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01240">Cleaner::removeColumns()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01262">Cleaner::removeSequences()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00267">setBlockSize()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00274">setKeepSequencesFlag()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="ad9df4d89e17aea55da3792967d4a430b"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ad9df4d89e17aea55da3792967d4a430b">&#9670;&nbsp;</a></span>dataType</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::dataType = <a class="el" href="defines_8h.html#a69d5409d6e7370d6bd3b4f931702e936a3391d45e4b0be480216715453e5d9ddf">SequenceTypes::NotDefined</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Flag variable that represents the type of the alignment. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00051">51</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>.</p>

</div>
</div>
<a id="aa0ab035fcc63d3245a93556b1280592f"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aa0ab035fcc63d3245a93556b1280592f">&#9670;&nbsp;</a></span>filename</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::filename</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Filename where this alignment was loaded from. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00084">84</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00087">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01884">trimAlManager::output_reports()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01668">trimAlManager::perform()</a>, <a class="el" href="_mold_8cpp_source.html#l00222">statistics::Mold::printAccumulative()</a>, <a class="el" href="_mold_8cpp_source.html#l00171">statistics::Mold::printByColumn()</a>, <a class="el" href="_similarity_8cpp_source.html#l00483">statistics::Similarity::printConservationAcl()</a>, <a class="el" href="_similarity_8cpp_source.html#l00423">statistics::Similarity::printConservationColumns()</a>, <a class="el" href="_gaps_8cpp_source.html#l00464">statistics::Gaps::printGapsAcl()</a>, <a class="el" href="_gaps_8cpp_source.html#l00387">statistics::Gaps::printGapsColumns()</a>, <a class="el" href="_consistency_8cpp_source.html#l00666">statistics::Consistency::printStatisticsFileAcl()</a>, <a class="el" href="_consistency_8cpp_source.html#l00606">statistics::Consistency::printStatisticsFileColumns()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00068">FormatHandling::FormatManager::replaceINtag()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01431">statSVG()</a>.</p>

</div>
</div>
<a id="aeb3c72eeb5d2954ad68369d48f8c16e6"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aeb3c72eeb5d2954ad68369d48f8c16e6">&#9670;&nbsp;</a></span>identities</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::identities = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>2D Matrix that represents how much each sequence resembles other sequences in the same MSA </p>
<p>Identity score between two sequences is the ratio of identical residues by the total length (common and no-common residues) among them. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00089">89</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01483">Cleaner::calculateRelaxedSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01525">Cleaner::calculateRepresentativeSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01068">Cleaner::getCutPointClusters()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00912">printSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00042">Cleaner::selectMethod()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="ae55f8a04d67451b23615467d24823daa"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae55f8a04d67451b23615467d24823daa">&#9670;&nbsp;</a></span>isAligned</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::isAligned = false</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Flag that indicates if all sequences on the alignment have the same length (Including gaps). </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00076">76</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00495">isFileAligned()</a>, and <a class="el" href="read_al_main_8cpp_source.html#l00359">main()</a>.</p>

</div>
</div>
<a id="a5754ea57c16c29a35a36243fa3746b70"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a5754ea57c16c29a35a36243fa3746b70">&#9670;&nbsp;</a></span>numberOfResidues</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::numberOfResidues = 0</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Number of residues present on the alignment if it is aligned. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00074">74</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01074">calculateColIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">calculateSeqOverlap()</a>, <a class="el" href="_similarity_8cpp_source.html#l00157">statistics::Similarity::calculateVectors()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01044">trimAlManager::check_select_cols_and_seqs_incompatibilities()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00231">Cleaner::cleanByCutValueFallBehind()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00090">Cleaner::cleanByCutValueOverpass()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00357">Cleaner::cleanByCutValueOverpassOrEquals()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00513">Cleaner::cleanStrict()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01589">Cleaner::computeComplementaryAlig()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00252">getNumAminos()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00868">printAlignmentInfo()</a>, <a class="el" href="_consistency_8cpp_source.html#l00666">statistics::Consistency::printStatisticsFileAcl()</a>, <a class="el" href="_consistency_8cpp_source.html#l00606">statistics::Consistency::printStatisticsFileColumns()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01326">Cleaner::removeSmallerBlocks()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00099">FormatHandling::FormatManager::saveAlignments()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01431">statSVG()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">updateSequencesAndResiduesNums()</a>.</p>

</div>
</div>
<a id="ad236d6ee5f8414dead5b9dabca27c291"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ad236d6ee5f8414dead5b9dabca27c291">&#9670;&nbsp;</a></span>numberOfSequences</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::numberOfSequences = 0</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Number of sequences present on the alignment. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00070">70</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_gaps_8cpp_source.html#l00181">statistics::Gaps::calcCutPoint()</a>, <a class="el" href="_gaps_8cpp_source.html#l00316">statistics::Gaps::calcCutPoint2ndSlope()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01074">calculateColIdentity()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">calculateSeqOverlap()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01044">trimAlManager::check_select_cols_and_seqs_incompatibilities()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00716">checkCorrespondence()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01589">Cleaner::computeComplementaryAlig()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01211">Cleaner::getClustering()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01068">Cleaner::getCutPointClusters()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00245">getNumSpecies()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00445">getSequenceNameOrder()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00416">getSequences()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">prepareCodingSequence()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00868">printAlignmentInfo()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00912">printSeqIdentity()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01001">printSeqOverlap()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00099">FormatHandling::FormatManager::saveAlignments()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00042">Cleaner::selectMethod()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">updateSequencesAndResiduesNums()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>, and <a class="el" href="_similarity_8cpp_source.html#l00075">statistics::Similarity::~Similarity()</a>.</p>

</div>
</div>
<a id="acb820d1138c4f0eda3d7c7d2ffe8fe66"></a>
<h2 class="memtitle"><span class="permalink"><a href="#acb820d1138c4f0eda3d7c7d2ffe8fe66">&#9670;&nbsp;</a></span>originalNumberOfResidues</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::originalNumberOfResidues = 0</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Number of residues the alignment had when it was loaded. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00072">72</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_gaps_8cpp_source.html#l00093">statistics::Gaps::applyWindow()</a>, <a class="el" href="_similarity_8cpp_source.html#l00271">statistics::Similarity::applyWindow()</a>, <a class="el" href="_gaps_8cpp_source.html#l00181">statistics::Gaps::calcCutPoint()</a>, <a class="el" href="_similarity_8cpp_source.html#l00381">statistics::Similarity::calcCutPoint()</a>, <a class="el" href="_gaps_8cpp_source.html#l00316">statistics::Gaps::calcCutPoint2ndSlope()</a>, <a class="el" href="_gaps_8cpp_source.html#l00226">statistics::Gaps::calcCutPointMixSlope()</a>, <a class="el" href="_similarity_8cpp_source.html#l00097">statistics::Similarity::calculateMatrixIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01483">Cleaner::calculateRelaxedSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00855">Cleaner::calculateSpuriousVector()</a>, <a class="el" href="_similarity_8cpp_source.html#l00157">statistics::Similarity::calculateVectors()</a>, <a class="el" href="_gaps_8cpp_source.html#l00593">statistics::Gaps::CalculateVectors()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00231">Cleaner::cleanByCutValueFallBehind()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00090">Cleaner::cleanByCutValueOverpass()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00357">Cleaner::cleanByCutValueOverpassOrEquals()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00980">Cleaner::cleanCombMethods()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00820">Cleaner::cleanCompareFile()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00513">Cleaner::cleanStrict()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01589">Cleaner::computeComplementaryAlig()</a>, <a class="el" href="_gaps_8cpp_source.html#l00038">statistics::Gaps::Gaps()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>, <a class="el" href="_mold_8cpp_source.html#l00222">statistics::Mold::printAccumulative()</a>, <a class="el" href="_mold_8cpp_source.html#l00171">statistics::Mold::printByColumn()</a>, <a class="el" href="_similarity_8cpp_source.html#l00483">statistics::Similarity::printConservationAcl()</a>, <a class="el" href="_similarity_8cpp_source.html#l00423">statistics::Similarity::printConservationColumns()</a>, <a class="el" href="_manager_8cpp_source.html#l00144">statistics::Manager::printCorrespondence()</a>, <a class="el" href="_gaps_8cpp_source.html#l00464">statistics::Gaps::printGapsAcl()</a>, <a class="el" href="_gaps_8cpp_source.html#l00387">statistics::Gaps::printGapsColumns()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01283">Cleaner::removeOnlyTerminal()</a>, <a class="el" href="sequences_matrix_8cpp_source.html#l00048">Alignment::sequencesMatrix::sequencesMatrix()</a>, <a class="el" href="_similarity_8cpp_source.html#l00040">statistics::Similarity::Similarity()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01431">statSVG()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">updateSequencesAndResiduesNums()</a>.</p>

</div>
</div>
<a id="a7edb0bea8267e7b9ac1b1171e14f9436"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a7edb0bea8267e7b9ac1b1171e14f9436">&#9670;&nbsp;</a></span>originalNumberOfSequences</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::originalNumberOfSequences = 0</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Number of sequences the alignment had when it was loaded. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00068">68</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_gaps_8cpp_source.html#l00093">statistics::Gaps::applyWindow()</a>, <a class="el" href="_gaps_8cpp_source.html#l00181">statistics::Gaps::calcCutPoint()</a>, <a class="el" href="_gaps_8cpp_source.html#l00226">statistics::Gaps::calcCutPointMixSlope()</a>, <a class="el" href="_similarity_8cpp_source.html#l00097">statistics::Similarity::calculateMatrixIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01483">Cleaner::calculateRelaxedSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01525">Cleaner::calculateRepresentativeSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00855">Cleaner::calculateSpuriousVector()</a>, <a class="el" href="_similarity_8cpp_source.html#l00157">statistics::Similarity::calculateVectors()</a>, <a class="el" href="_gaps_8cpp_source.html#l00593">statistics::Gaps::CalculateVectors()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01308">trimAlManager::check_absolute_gap_theshold()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01049">Cleaner::cleanNoAllGaps()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00714">Cleaner::cleanOverlapSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00934">Cleaner::cleanSpuriousSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01589">Cleaner::computeComplementaryAlig()</a>, <a class="el" href="_gaps_8cpp_source.html#l00038">statistics::Gaps::Gaps()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01211">Cleaner::getClustering()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01068">Cleaner::getCutPointClusters()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01668">trimAlManager::perform()</a>, <a class="el" href="_gaps_8cpp_source.html#l00464">statistics::Gaps::printGapsAcl()</a>, <a class="el" href="_gaps_8cpp_source.html#l00387">statistics::Gaps::printGapsColumns()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00912">printSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01610">Cleaner::removeDuplicates()</a>, <a class="el" href="sequences_matrix_8cpp_source.html#l00048">Alignment::sequencesMatrix::sequencesMatrix()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01431">statSVG()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">updateSequencesAndResiduesNums()</a>.</p>

</div>
</div>
<a id="a1b63fac4259b20561bf46369d8908d82"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a1b63fac4259b20561bf46369d8908d82">&#9670;&nbsp;</a></span>overlaps</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::overlaps = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Matrix that stores the overlap between sequences of an alignment. See <a class="el" href="class_alignment.html#ab4d73aec4e1ec82bee55321791a1c6dc" title="Calculates overlap between sequences. ">Alignment::calculateSeqOverlap</a>. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00092">92</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">calculateSeqOverlap()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01001">printSeqOverlap()</a>.</p>

</div>
</div>
<a id="ad8384252e067c742dc5d249b34925891"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ad8384252e067c742dc5d249b34925891">&#9670;&nbsp;</a></span>saveResidues</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::saveResidues = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Vector containing which residues will be kept/removed.<br />
-1 Indicates a 'removed/toRemove'. Otherwise residue/column will be kept. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00095">95</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01483">Cleaner::calculateRelaxedSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00231">Cleaner::cleanByCutValueFallBehind()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00090">Cleaner::cleanByCutValueOverpass()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00357">Cleaner::cleanByCutValueOverpassOrEquals()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00980">Cleaner::cleanCombMethods()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00513">Cleaner::cleanStrict()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01589">Cleaner::computeComplementaryAlig()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="_manager_8cpp_source.html#l00144">statistics::Manager::printCorrespondence()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01240">Cleaner::removeColumns()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01283">Cleaner::removeOnlyTerminal()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01326">Cleaner::removeSmallerBlocks()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">updateSequencesAndResiduesNums()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="a415ed54c53c4775263659e6d8a25e998"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a415ed54c53c4775263659e6d8a25e998">&#9670;&nbsp;</a></span>saveSequences</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::saveSequences = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Vector containing which sequences will be kept/removed.<br />
-1 Indicates a 'removed/toRemove'. Otherwise sequence will be kept. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00096">96</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01483">Cleaner::calculateRelaxedSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01525">Cleaner::calculateRepresentativeSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_gaps_8cpp_source.html#l00593">statistics::Gaps::CalculateVectors()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00714">Cleaner::cleanOverlapSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01589">Cleaner::computeComplementaryAlig()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01211">Cleaner::getClustering()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01068">Cleaner::getCutPointClusters()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01610">Cleaner::removeDuplicates()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01262">Cleaner::removeSequences()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l02347">updateSequencesAndResiduesNums()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="a5373ff2b4fbeea7875a4613758276518"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a5373ff2b4fbeea7875a4613758276518">&#9670;&nbsp;</a></span>SeqRef</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::SeqRef = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Pointer to keep a count of how many alignments depend on the same shared-sequences.<br />
 Increased on each trim step, and automatically reduced on each alignment destroy.<br />
 If it arrives to 0, the last alignment being destroyed will also take care of releasing memory for shared information. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00066">66</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="abe93ddf27b0ef2c3cb7f4d0232a4d675"></a>
<h2 class="memtitle"><span class="permalink"><a href="#abe93ddf27b0ef2c3cb7f4d0232a4d675">&#9670;&nbsp;</a></span>seqsInfo</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::seqsInfo = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Vector containing extra information about each sequence. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00082">82</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="afae240350928bf95b01456cbeb631767"></a>
<h2 class="memtitle"><span class="permalink"><a href="#afae240350928bf95b01456cbeb631767">&#9670;&nbsp;</a></span>seqsName</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::seqsName = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>String vector containing the sequences names. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00080">80</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00716">checkCorrespondence()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00445">getSequenceNameOrder()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00416">getSequences()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01668">trimAlManager::perform()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">prepareCodingSequence()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00868">printAlignmentInfo()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00912">printSeqIdentity()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01001">printSeqOverlap()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01610">Cleaner::removeDuplicates()</a>, <a class="el" href="sequences_matrix_8cpp_source.html#l00048">Alignment::sequencesMatrix::sequencesMatrix()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="ab0cafdefe306c19917a8c58abd2f8918"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ab0cafdefe306c19917a8c58abd2f8918">&#9670;&nbsp;</a></span>sequences</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::sequences = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Vector containing the sequences. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00078">78</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01074">calculateColIdentity()</a>, <a class="el" href="_manager_8cpp_source.html#l00166">statistics::Manager::calculateGapStats()</a>, <a class="el" href="_similarity_8cpp_source.html#l00097">statistics::Similarity::calculateMatrixIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01483">Cleaner::calculateRelaxedSeqIdentity()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01525">Cleaner::calculateRepresentativeSeq()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01434">Cleaner::calculateSeqIdentity()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00378">calculateSeqOverlap()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00855">Cleaner::calculateSpuriousVector()</a>, <a class="el" href="_similarity_8cpp_source.html#l00157">statistics::Similarity::calculateVectors()</a>, <a class="el" href="_gaps_8cpp_source.html#l00593">statistics::Gaps::CalculateVectors()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00716">checkCorrespondence()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00794">fillMatrices()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00469">getAlignmentType()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01068">Cleaner::getCutPointClusters()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00424">getSequences()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00169">getTranslationCDS()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00528">prepareCodingSequence()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00868">printAlignmentInfo()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01367">Cleaner::removeAllGapsSeqsAndCols()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01610">Cleaner::removeDuplicates()</a>, <a class="el" href="sequences_matrix_8cpp_source.html#l00048">Alignment::sequencesMatrix::sequencesMatrix()</a>, <a class="el" href="_base_format_handler_8cpp_source.html#l00267">FormatHandling::FormatManager::splitAlignmentKeeping()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="a032a37b570890ebfb3a925c329c36b43"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a032a37b570890ebfb3a925c329c36b43">&#9670;&nbsp;</a></span>SequencesMatrix</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::SequencesMatrix = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>SequencesMatrix submodule<br />
 This helper module stores the sequences without gaps. Useful when comparing MSA. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00062">62</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_consistency_8cpp_source.html#l00241">statistics::Consistency::compareAndChoose()</a>, <a class="el" href="_consistency_8cpp_source.html#l00416">statistics::Consistency::forceComparison()</a>, <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<a id="a26d027409ce208953560d3e5916ad616"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a26d027409ce208953560d3e5916ad616">&#9670;&nbsp;</a></span>Statistics</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Alignment::Statistics = nullptr</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Statistics submodule.<br />
It contains methods and variables related to statistics calculation and reporting. </p>

<p class="definition">Definition at line <a class="el" href="_alignment_8h_source.html#l00060">60</a> of file <a class="el" href="_alignment_8h_source.html">Alignment.h</a>.</p>

<p class="reference">Referenced by <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00046">Alignment()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01167">alignmentSummaryHTML()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01708">alignmentSummarySVG()</a>, <a class="el" href="_manager_8cpp_source.html#l00036">statistics::Manager::calculateConservationStats()</a>, <a class="el" href="_similarity_8cpp_source.html#l00157">statistics::Similarity::calculateVectors()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00784">Cleaner::clean()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00957">Cleaner::clean2ndSlope()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00980">Cleaner::cleanCombMethods()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00760">Cleaner::cleanConservation()</a>, <a class="el" href="_cleaner_8cpp_source.html#l00736">Cleaner::cleanGaps()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01049">Cleaner::cleanNoAllGaps()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02145">trimAlManager::CleanResiduesNonAuto()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l02051">trimAlManager::CleanSequences()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01963">trimAlManager::create_or_use_similarity_matrix()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01803">trimAlManager::innerPerform()</a>, <a class="el" href="_consistency_8cpp_source.html#l00045">statistics::Consistency::perform()</a>, <a class="el" href="trimal_manager_8cpp_source.html#l01915">trimAlManager::print_statistics()</a>, <a class="el" href="_manager_8cpp_source.html#l00073">statistics::Manager::printStatisticsConservationColumns()</a>, <a class="el" href="_manager_8cpp_source.html#l00086">statistics::Manager::printStatisticsConservationTotal()</a>, <a class="el" href="_manager_8cpp_source.html#l00118">statistics::Manager::printStatisticsGapsColumns()</a>, <a class="el" href="_manager_8cpp_source.html#l00131">statistics::Manager::printStatisticsGapsTotal()</a>, <a class="el" href="_cleaner_8cpp_source.html#l01283">Cleaner::removeOnlyTerminal()</a>, <a class="el" href="_manager_8cpp_source.html#l00100">statistics::Manager::setSimilarityMatrix()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00259">setWindowsSize()</a>, <a class="el" href="_alignment_2_alignment_8cpp_source.html#l01431">statSVG()</a>, and <a class="el" href="_alignment_2_alignment_8cpp_source.html#l00134">~Alignment()</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li>/home/vfernandez/git/MyFork/include/Alignment/<a class="el" href="_alignment_8h_source.html">Alignment.h</a></li>
<li>/home/vfernandez/git/MyFork/source/Alignment/<a class="el" href="_alignment_2_alignment_8cpp_source.html">Alignment.cpp</a></li>
<li>/home/vfernandez/git/MyFork/Documentation/ClassDocumentation/<a class="el" href="_alignment_8dox.html">Alignment.dox</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="class_alignment.html">Alignment</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
